src/main/java/io/github/thebusybiscuit/slimefun4/utils/BlockUtils.java

```diff
@@ -0,0 +1,32 @@
+package io.github.thebusybiscuit.slimefun4.utils;
+import io.github.thebusybiscuit.cscorelib2.reflection.ReflectionUtils;
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+public final class BlockUtils {
+
+    private static final boolean is_1_14 = ReflectionUtils.isVersion("v1_14_");
+
+    private BlockUtils() {}
+
+    public static boolean hasInventory(Block block) {
+        if (block == null) return false;
+
+        Material type = block.getType();
+        switch (type) {
+            case CHEST:
+            case TRAPPED_CHEST:
+            case FURNACE:
+            case DISPENSER:
+            case DROPPER:
+            case HOPPER:
+            case BREWING_STAND:
+            case ENDER_CHEST:
```

Comment:
I don't think Ender Chests have Inventories that you can interact with...
Since those are per-player not per-block.